app/src/main/java/com/asma/tasky/core/presentation/components/Navigation.kt

```diff
@@ -49,13 +56,64 @@ fun Navigation(
         }
 
         composable(Screen.TaskScreen.route) {
-            TaskScreen()
+            // todo check if a task object is passed from Agenda Screen
+            // todo also check if should show in edit mode
+
+            var title = stringResource(R.string.new_task)
+            var description = stringResource(R.string.task_description)
+
+            if (navController.currentBackStackEntry!!.savedStateHandle.contains(Constants.KEY_TITLE)) {
+                title =
+                    navController.currentBackStackEntry!!.savedStateHandle.get<String>(
+                        Constants.KEY_TITLE
+                    ) ?: stringResource(R.string.new_task)
+            }
```

Comment:
This if-check seems unnecessary, since `savedStateHandle.get()` will already return null if the value doesn't exist.